sphinx/source/docs/dev_guide/bokehjs.rst

```diff
@@ -121,98 +121,226 @@ Use ``node make help`` to list all available commands.
+    few elements as possible on your plot.
+
+    Follow this general pattern for visual tests:
+
+    .. code-block:: TypeScript
+
+      describe("Object", () => {
+        it("should show certain behavior", async () => {
+          const p = fig([width, height])
+
+          ...
+
+          await display(p)
+        })
+
+2. Run tests locally:
+    Run ``node make tests`` to test your changes on your system. To only run integration
+    tests, use ``node make test:integration``.
+
+    If you want to only run a specific test, use the ``-k`` argument and supply a search
+    string. The search string is case sensitive. The BokehJS testing suite tries to
+    match your search string to the strings defined in the code's ``it()`` and
+    ``should()`` functions. For example:
+
+    .. code-block:: sh
+
+      $ node make test:integration -k 'Legend annotation'
+
+    The first time you run a new or updated visual test, the BokehJS testing framework
+    will notify you that baseline files are missing our outdated. At this point, it will
+    also generate all missing or outdated baseline files for your operating system. The
+    baseline files will be in a subfolder of :bokeh-tree:`bokehjs/test/baselines/`.
+
+    You can use any png viewer to inspect the generated png files. Adjust your testing
+    code until the test's visual output matches your expectations.
+
+    Optionally, you can also use the BokehJS
+    :ref:`devtools server <devguide_bokehjs_development_devtoolsserver>` to review
+    your local test results.
+
+3. Generate CI baselines and commit test:
+    As a final step before pushing your visual tests to Bokeh's GitHub repository, you
+    need to generate and commit the baseline files using
+    :ref:`Bokeh's CI <devguide_testing_ci>`.
+
+    [TBD]
+
+    Push your changes to GitHub and wait for CI to finish.
+    If you added new tests, CI will expectedly fail with "missing baseline
+    images" error message.
+    If tests passed then you are done.
+    If tests failed, go to BokehJS's GitHub_Actions_ page. Find the most recent
+    test run for your PR and download the associated ``bokehjs-report`` artifact.
+    Unzip the artifact archive at the root of the repository.
+    Assuming :ref:`devtools server <devguide_bokehjs_development_devtoolsserver>` is running in the background, go to ``/integration/report?platform=name``
+    where ``name`` is either ``linux``, ``macos`` or ``windows`` and review the test output
+    for each platform. If there are no unintentional differences, then commit all
+    new or modified ``*.blf`` and ``*.png`` files under ``test/baselines/{linux,macos,windows}``.
+    Push your changes to GitHub again and verify that tests pass this time.
+
+    .. warning::
+      Make sure to only push baseline files to the CI that were created by the CI. Do
+      not include any locally created baseline files in your pull request.
```

Comment:
I think you mean baseline images here, because it's conflicting with line 289 where its stated to commit the baseline files